components/camera/client_test.go

```diff
@@ -272,63 +265,52 @@ func TestClient(t *testing.T) {
 		camClient, err := camera.NewClientFromConn(context.Background(), conn, "", camera.Named(testCameraName), logger)
 		test.That(t, err, test.ShouldBeNil)
 
-		injectCamera.StreamFunc = func(ctx context.Context, errHandlers ...gostream.ErrorHandler) (gostream.VideoStream, error) {
-			extra, ok := camera.FromContext(ctx)
-			test.That(t, ok, test.ShouldBeTrue)
+		injectCamera.ImageFunc = func(ctx context.Context, mimeType string, extra map[string]interface{}) ([]byte, camera.ImageMetadata, error) {
 			test.That(t, extra, test.ShouldBeEmpty)
-			return nil, errStreamFailed
+			return nil, camera.ImageMetadata{}, errGetImageFailed
 		}
 
 		ctx := context.Background()
-		_, _, err = camera.ReadImage(ctx, camClient)
+		_, _, err = camClient.Image(ctx, "", nil)
 		test.That(t, err, test.ShouldNotBeNil)
-		test.That(t, err.Error(), test.ShouldContainSubstring, errStreamFailed.Error())
+		test.That(t, err.Error(), test.ShouldContainSubstring, errGetImageFailed.Error())
```

Comment:
please assert on the actual error, not the substring

Reply:
```
    --- FAIL: TestClient/camera_client_extra (0.00s)
        /Users/seanyu/Projects/sean-rdk/components/camera/client_test.go:288: Expected: 'can't get image'
            Actual:   'rpc error: code = Unknown desc = can't get image'
            (Should be equal)
```
Getting this when trying to change this line to use `test.ShouldEqual` for both the expected and actual value. Looks like it's an RPC error that wraps `errGetImageFailed`. Instinctively after seeing this, I would've gone down the substring route anyways... is there a workaround here to maybe unwrap it and compare?

Reply:
Same deal as https://github.com/viamrobotics/rdk/pull/4487#discussion_r1843887777